File: tests/Generator/resource/csharp/complex/BooleanPropertyType.cs
using System.Text.Json.Serialization;

namespace TypeAPI.Model;

/// <summary>
/// Represents a boolean value
/// </summary>
public class BooleanPropertyType : ScalarPropertyType
{
    [JsonPropertyName("type")]
    public new string? Type { get; set; } = "boolean";

}

File: tests/Generator/resource/java/complex/BooleanPropertyType.java
package org.typeapi.model;

import com.fasterxml.jackson.annotation.*;

@JsonClassDescription("Represents a boolean value")
public class BooleanPropertyType extends ScalarPropertyType {
    private String type = "boolean";

    @JsonSetter("type")
    public void setType(String type) {
        this.type = type;
    }

    @JsonGetter("type")
    public String getType() {
        return this.type;
    }
}

